### PR TITLE
🐛 Add regex validation for signer

### LINKED ISSUE #382: Should validate the `registration.customSigner.signerName` in the addontemplate
**Describe the bug**
If I used a invalid signerName, e.g. `xxx-addon`, there are no any hints  to specify this, unless checking the registration agent log

```
Invalid value: "xxx-addon": must be a fully qualified domain and path of the form 'example.com/signer-name'
```

**To Reproduce**
```
 registration:
  - type: CustomSigner
    customSigner:
      signerName: xxx-addon
      signingCA:
        name: xxxx
```

**Expected behavior**
Validate this when I create the addontemplate

### PR DESCRIPTION
**Pull request by qiujian16**
## Related issue(s)

Fixes https://github.com/open-cluster-management-io/ocm/issues/382

### CODE REVIEW

**Comment by qiujian16** [comment]
/hold
will add integration test.

/assign @zhujian7

File: addon/v1alpha1/0000_01_addon.open-cluster-management.io_managedclusteraddons.crd.yaml [hunk 1]
```diff
@@ -314,6 +314,7 @@ spec:
                         will use to create csr.
                       maxLength: 571
                       minLength: 5
+                      pattern: ^([a-z0-9][a-z0-9-]+\.)+[a-z]+\/[a-z0-9-\.]+$
```
Comment (zhujian7) [inline addon/v1alpha1/0000_01_addon.open-cluster-management.io_managedclusteraddons.crd.yaml]:
```suggestion
                      pattern: ^([a-z0-9][a-z0-9-]*\.)+[a-z]+\/[a-z0-9-\.]+$
```

**Pull request merged**